bot/exts/filters/token_remover.py

```diff
@@ -116,6 +124,50 @@ async def take_action(self, msg: Message, found_token: Token) -> None:
 
         self.bot.stats.incr("tokens.removed_tokens")
 
+    async def handle_mfa_token(self, msg: Message) -> None:
+        """
+        Check all messages for a string that matches the mfa token pattern.
+
+        Due to how the mfa tokens work, there is no user information supplied for this token.
+        """
+        # Ignore DMs; can't delete messages in there anyway.
+        if not msg.guild or msg.author.bot:
+            return
+
+        was_valid = False
+        for match in MFA_TOKEN_RE.finditer(msg.content):
+
+            if self.is_maybe_valid_hmac(match.group()):
```

Comment:
`match.group()` here will include `mfa.` so this will always return true, only the characters after should be passed here.